libpromises/expand.c

```diff
@@ -962,16 +962,22 @@ static void ResolveControlBody(EvalContext *ctx, GenericAgentConfig *config,
         {
             Log(LOG_LEVEL_VERBOSE, "SET ignore_missing_inputs %s",
                 RvalScalarValue(evaluated_rval));
-            config->ignore_missing_inputs = BooleanFromString(
-                RvalScalarValue(evaluated_rval));
+            if (StringIsBoolean(RvalScalarValue(evaluated_rval)))
+            {
+                config->ignore_missing_inputs = BooleanFromString(
```

Comment:
So this is initialized to `false`?

Reply:
Yes, it is initialized to false and documented that false is the default. We may have to add some tricks, either to the tests, or to the C code, to make it default to false at a later point, for example after pre-eval or after some evaluation passes. Need to test more.